wg-api/spec-documents/scroll-view.md

```diff
@@ -0,0 +1,281 @@
+# Scroll View
+
+## Summary
+
+This spec proposes to add an experimental native ScrollView component to Electron.
+
+## Platforms
+* macOS 
+* Windows
+* Linux
+
+## Impetus
+
+It is currently not possible to add scrolling functionality around BrowserViews:
+
+* Capture mouse / touch events
+* Reposition BrowserViews accordingly
+
+More details of the use-case and challenges related to workarounds: [#32751](https://github.com/electron/electron/issues/32751)
+
+![](https://user-images.githubusercontent.com/2891206/150981006-84fe796e-60bd-43c0-952d-831d1bfd9998.gif)
+
+## Usage Example
+
+```js
+const { BrowserView, BrowserWindow, BaseView, ScrollView, WebBrowserView } = require("electron");
+
+const window = new BrowserWindow({ width: 1000, height: 500 })
+
+const scroll = new ScrollView()
+scroll.setBounds({ width: 1000, height: 500 })
+
+const scrollContent = new BaseView()
+scrollContent.setBounds({ width: 2000, height: 500 })
+
+const webBrowserView1 = new WebBrowserView()
+webBrowserView1.setBounds({ width: 1000, height: 500 })
+scrollContent.addChild(webBrowserView1)
+
+const webBrowserView2 = new WebBrowserView()
+webBrowserView2.setBounds({ width: 1000, height: 500, x: 1000 })
+scrollContent.addChild(webBrowserView2)
+
+scroll.setContentView(scrollContent);
+win.addChild(scroll);
+```
+
+
+## API Design
+
+In order to achieve the example above, we've made following changes to the API
+
+## BaseView
+
+A `BaseView` is a rectangle within the views View hierarchy. It is the base
+class for all all different views. 
```

Comment:
What's the relationship between this and the existing `View` class?

Reply:
I couldn't make BaseView and View compatible - because COCOA used for BaseView (on Mac). So there is no relationship.

Reply:
Wait, what cocoa-specific features are in BaseView? I would think those could be integrated into View. We should NOT have two _different_ View base classes.

Reply:
At https://github.com/electron/electron/issues/32751#issuecomment-1034811577 we were discussing with @zcbenz this topic, he agreed for the possibility  of use NSView on macOS and views::View on others.
There are several reasons why I have designed the current implementation in this direction:
- vibrancy effect on Mac for views,
- animations for setting the bounds, scaling etc. easily implementable in Cocoa,
- some other visual properties, like shapes for NSView;
Most of these features can also be achieved with Chromium views, but probably not all.

Reply:
I don't think we want to expose any NSView stuff directly. If we need extra features from views that views doesn't already support, I think it could make sense to patch those things in or otherwise extend views to support them. But I think it is a bad idea to have one API that backs onto views on win/linux and onto NSView/Cocoa on mac. Chrome used to do this and moved away from it.

Views [supports animations](https://source.chromium.org/chromium/chromium/src/+/main:ui/views/animation/); I'm not sure what you mean by "shaping" but `View::SetClipPath` is a thing. Vibrancy isn't supported by views, but it seems like something we could build on top of views if needed. I don't think it calls for a whole separate implementation.

Reply:
There is a lot of bugs with the Chromium's ScrollView, for example UiCompositorScrollWithLayers (which provides the dezired performance for WebViews) is implemented only on MacOs. But currently we're working on it, we already have a ready solution for Windows, we also implement other features (to be consistent with Mac) - such as elasticity. So we agree that in this situation the best solution is to use Chromium views on all platforms - and implement the missing features.

Reply:
Fixed in 8418e0413a4cccc2258add477052713d647a9300